src/main/java/plugily/projects/villagedefense/commands/arguments/game/LeaderboardArgument.java

```diff
@@ -78,32 +72,20 @@ public void execute(CommandSender sender, String[] args) {
   }
 
   private void printLeaderboard(CommandSender sender, StatsStorage.StatisticType statisticType) {
-    java.util.Map<UUID, Integer> stats = StatsStorage.getStats(statisticType);
+    Map<UUID, Integer> stats = StatsStorage.getStats(statisticType);
     sender.sendMessage(registry.getPlugin().getChatManager().colorMessage(Messages.LEADERBOARD_HEADER));
     String statistic = StringUtils.capitalize(statisticType.toString().toLowerCase().replace('_', ' '));
-    for(int i = 0; i < 10; i++) {
-      Object[] array = stats.keySet().toArray();
-      try {
-        UUID current = (UUID) array[array.length - 1];
-        sender.sendMessage(formatMessage(statistic, Bukkit.getOfflinePlayer(current).getName(), i + 1, stats.get(current)));
-        stats.remove(current);
-      } catch(IndexOutOfBoundsException ex) {
-        sender.sendMessage(formatMessage(statistic, "Empty", i + 1, 0));
-      } catch(NullPointerException ex) {
-        UUID current = (UUID) array[array.length - 1];
-        if(registry.getPlugin().getConfigPreferences().getOption(ConfigPreferences.Option.DATABASE_ENABLED)) {
-          try(Connection connection = registry.getPlugin().getMysqlDatabase().getConnection();
-              Statement statement = connection.createStatement();
-              ResultSet set = statement.executeQuery("SELECT name FROM " + ((MysqlManager) registry.getPlugin().getUserManager().getDatabase()).getTableName() + " WHERE UUID='" + current.toString() + "'")) {
-            if(set.next()) {
-              sender.sendMessage(formatMessage(statistic, set.getString(1), i + 1, stats.get(current)));
-              continue;
-            }
-          } catch(SQLException ignored) {
-            //it has failed second time, cannot continue
-          }
+    UUID[] array = stats.keySet().toArray(new UUID[0]);
```

Comment:
`array` length will be 0 as you given 0 length for the array parameter. You need to specify `stats.keySet` method size.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/List.html#toArray-T:A-

Reply:
🤦‍♂️ 🤦‍♂️ 🤦‍♂️ <- me